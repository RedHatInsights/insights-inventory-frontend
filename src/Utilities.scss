// Sass Imports

@import 'Utilities/_variables.scss';
@import 'Utilities/_colors.scss';
@import 'Utilities/_helpers.scss';
@import 'Utilities/_mixins.scss';

@import '~@redhat-cloud-services/frontend-components/index.css';
