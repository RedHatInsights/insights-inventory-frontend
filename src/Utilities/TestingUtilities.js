import React from 'react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';

export const mountWithRouter = (Component, initialEntries) => {
  const wrapper = mount(
    <MemoryRouter initialEntriest={initialEntries}>{Component}</MemoryRouter>
  );

  return wrapper;
};

export const TestWrapper = ({
  children,
  routerProps = { initialEntries: ['/'] },
  path,
  store = {},
}) => {
  return (
    <MemoryRouter {...routerProps}>
      <Provider store={store}>
        {path ? (
          <Routes>
            <Route path={path} element={children} />
          </Routes>
        ) : (
          children
        )}
      </Provider>
    </MemoryRouter>
  );
};

TestWrapper.propTypes = {
  children: PropTypes.any.isRequired,
  routerProps: PropTypes.shape({
    initialEntries: PropTypes.arrayOf(PropTypes.string),
  }),
  path: PropTypes.string,
  store: PropTypes.any,
};
