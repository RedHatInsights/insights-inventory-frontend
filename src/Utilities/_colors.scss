// Sample Colors
$light: var(--pf-global--BackgroundColor--300);
$medium: #393f44;
$dark: #292e34;

$primary: #39a5dc;
$primary-dark: #0088ce;
$primary-shadow: #00659c;

$accent: #2A2E57;
$accent-dark: #20253b;
