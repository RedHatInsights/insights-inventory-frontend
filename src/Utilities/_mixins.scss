@use 'sass:meta';
@use 'sass:math';
@use 'sass:list';
@use 'sass:string';

// Convert px to rem
// =================================================================/
@mixin rem($property, $values) {

  // Create a couple of empty lists as output buffers.
  $font-size: 14px;
  $px-values: ();
  $rem-values: ();

  // Loop through the $values list
  @each $value in $values {

    // For each property value, if it's in rem or px, derive both rem and
    // px values for it and add those to the end of the appropriate buffer.
    // Ensure all pixel values are rounded to the nearest pixel.
    @if meta.type-of($value) == number and not math.is-unitless($value) and (math.unit($value) == px) {

      // px value given - calculate rem value from font-size
      $new-rem-value: calc($value / $font-size);
      $px-values: list.join($px-values, round($value));
      $rem-values: list.join($rem-values, string.unquote("#{$new-rem-value}rem"));

    } @else if meta.type-of($value) == number and not math.is-unitless($value) and (math.unit($value) == '%') {

      // % value given - don't add px or rem
      $px-values: list.join($px-values, string.unquote(#{$value}));
      $rem-values: list.join($rem-values, string.unquote(#{$value}));

    } @else if $value == auto {

      // auto - don't add px or rem
      $px-values: list.join($px-values, auto);
      $rem-values: list.join($rem-values, auto);

    } @else {

      // math.is-unitless value - use those directly as rem and calculate the px-fallback
      $px-values: list.join($px-values, round($value * $font-size));
      $rem-values: list.join($rem-values, string.unquote("#{$value}rem"));
    }
  }

  // output the converted rules
  #{$property}: $px-values;
  #{$property}: $rem-values;

}



// Generate font-size in rem
// =================================================================/
@mixin font-size($size, $line: $size * 1.4) {
  font-size: ($size);
  font-size: ($size / $global-font-size) + rem;

  @if $line == 1 {
    line-height: 1;

  } @else if $line != null {
    line-height: $line;
    line-height: ($line / $global-font-size) + rem;
  }
}



// Generate transition
// =================================================================/
@mixin transition($property) {

  transition: $property 0.2s cubic-bezier(0.4, 0, 0.2, 1);
  -o-transition: $property 0.2s cubic-bezier(0.4, 0, 0.2, 1);
  -ms-transition: $property 0.2s cubic-bezier(0.4, 0, 0.2, 1);
  -moz-transition: $property 0.2s cubic-bezier(0.4, 0, 0.2, 1);
  -webkit-transition: $property 0.2s cubic-bezier(0.4, 0, 0.2, 1);
}