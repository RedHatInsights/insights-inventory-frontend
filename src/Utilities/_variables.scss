// Global Variables
$global-padding: 30px;
$global-margin: 15px;

$global-font-size: 14px;

// PF Font
$pf-v5-global--enable-font-overpass-cdn: true;