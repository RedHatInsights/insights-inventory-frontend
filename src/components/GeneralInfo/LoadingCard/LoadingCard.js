import React, { Fragment, useEffect } from 'react';
import PropTypes from 'prop-types';
import {
    Stack,
    StackItem,
    TextContent,
    Text,
    TextList,
    TextVariants,
    TextListItemVariants,
    TextListVariants,
    TextListItem,
    Card,
    CardBody
} from '@patternfly/react-core';
import { Skeleton, SkeletonSize } from '@redhat-cloud-services/frontend-components/Skeleton';
import { useLocation, useHistory } from 'react-router-dom';

const valueToText = (value, singular, plural) => {
    if ((value || value === 0) && singular) {
        return value === 1 ? `1 ${singular}` : `${value} ${plural || `${singular}s`}`;
    }

    if (value === 0) {
        return 'None';
    }

    return value || 'Not available';
};

export const Clickable = ({ value, target, plural, singular, onClick }) => {
    const history = useHistory();
    const { pathname } = useLocation();
    const modalId = pathname.split('/').pop();
    useEffect(() => {
        if (target === modalId) {
            onClick({ value, target });
        }
    }, [modalId, target]);
    return (
        <a
            onClick={ event => {
                event.preventDefault();
                history.push(`${pathname}/${target}`);
            } }
            href={ `${window.location.origin}${window.location.pathname}/${target}` }
        >
            { valueToText(value, singular, plural) }
        </a>
    );
};

Clickable.propTypes = {
    value: PropTypes.node,
    target: PropTypes.string,
    onClick: PropTypes.func,
    plural: PropTypes.string,
    singular: PropTypes.string
};

const LoadingCard = ({ title, isLoading, items, children }) => (
    <Card>
        <CardBody>
            <Stack hasGutter>
                <StackItem>
                    <TextContent>
                        <Text component={TextVariants.h1}>
                            {title}
                        </Text>
                    </TextContent>
                </StackItem>
                <StackItem isFilled>
                    {items.length ?
                        (<TextContent>
                            <TextList component={TextListVariants.dl}>
                                {items.map(({ onClick, value, target, plural, singular, size, title: itemTitle }, key) => (
                                    <Fragment key={key}>
                                        <TextListItem component={TextListItemVariants.dt}>
                                            {itemTitle}
                                        </TextListItem>
                                        <TextListItem component={TextListItemVariants.dd}>
                                            {isLoading && <Skeleton size={size || SkeletonSize.sm} />}
                                            {!isLoading && (
                                                onClick && value ?
                                                    <Clickable
                                                        onClick={onClick}
                                                        value={value}
                                                        target={target}
                                                        plural={plural}
                                                        singular={singular}
                                                    /> :
                                                    valueToText(value, singular, plural)
                                            )}
                                        </TextListItem>
                                    </Fragment>
                                ))}
                            </TextList>
                        </TextContent>) : null}
                    {children}
                </StackItem>
            </Stack>
        </CardBody>

    </Card>

);

LoadingCard.propTypes = {
    title: PropTypes.node.isRequired,
    isLoading: PropTypes.bool,
    items: PropTypes.arrayOf(PropTypes.shape({
        title: PropTypes.node,
        value: PropTypes.node,
        onClick: PropTypes.func,
        size: PropTypes.oneOf(Object.values(SkeletonSize)),
        plural: PropTypes.string,
        singular: PropTypes.string
    })),
    children: PropTypes.node
};

LoadingCard.defaultProps = {
    isLoading: true,
    items: []
};

export default LoadingCard;
