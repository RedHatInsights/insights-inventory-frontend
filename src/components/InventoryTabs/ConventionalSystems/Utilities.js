import flatMap from 'lodash/flatMap';
import {
  HOST_GROUP_CHIP,
  RHCD_FILTER_KEY,
  UPDATE_METHOD_KEY,
} from '../../../Utilities/constants';

const mapTags = ({ category, values }) =>
  values.map(
    ({ tagKey, value }) =>
      `${category ? `${category}/` : ''}${tagKey}${value ? `=${value}` : ''}`
  );

const filterMapper = {
  staleFilter: ({ staleFilter }, searchParams) =>
    staleFilter.forEach((item) => searchParams.append('status', item)),
  osFilter: ({ osFilter }, searchParams) => {
    // TODO This is very hackish. There should be a schema/feature on how the values translate into params
    const osParams = Object.entries(
      Object.values(osFilter).reduce((acc, item) => ({ ...acc, ...item }), {})
    )
      .filter(([, value]) => value === true)
      .map(([key]) => key);

    osParams.forEach((item) => {
      const keyParts = item.split('-');
      const [osName, value] = [
        keyParts.slice(0, keyParts.length - 2).join(' '),
        keyParts[keyParts.length - 1],
      ];

      if (value && osName) {
        searchParams.append('operating_system', `${osName}${value}`);
      }
    });
  },
  registeredWithFilter: ({ registeredWithFilter }, searchParams) =>
    registeredWithFilter?.forEach((item) =>
      searchParams.append('source', item)
    ),
  value: ({ value, filter }, searchParams) =>
    value === 'hostname_or_id' &&
    Boolean(filter) &&
    searchParams.append('hostname_or_id', filter),
  tagFilters: ({ tagFilters }, searchParams) =>
    tagFilters?.length > 0 &&
    searchParams.append('tags', flatMap(tagFilters, mapTags)),
  rhcdFilter: ({ rhcdFilter }, searchParams) =>
    rhcdFilter?.forEach((item) => searchParams.append(RHCD_FILTER_KEY, item)),
  lastSeenFilter: ({ lastSeenFilter }, searchParams) =>
    Object.keys(lastSeenFilter || {})?.forEach(
      (item) =>
        item === 'mark' &&
        searchParams.append('last_seen', lastSeenFilter[item])
    ),
  updateMethodFilter: ({ updateMethodFilter }, searchParams) =>
    updateMethodFilter?.forEach((item) =>
      searchParams.append(UPDATE_METHOD_KEY, item)
    ),
  hostGroupFilter: ({ hostGroupFilter }, searchParams) =>
    hostGroupFilter?.forEach((item) =>
      searchParams.append(HOST_GROUP_CHIP, item)
    ),
};

// TODO It might be better to not use a SearchParam instance and mutate it
// This should rather just return a plain object that can be
// applied with react routers useSearch params.
export const calculateFilters = (searchParams, filters = []) => {
  filters.forEach((filter) => {
    Object.keys(filter).forEach((key) => {
      filterMapper?.[key]?.(filter, searchParams);
    });
  });
  return searchParams;
};

export const calculatePagination = (searchParams, page, perPage) => {
  const currSearch = new URLSearchParams(location.search);
  const newPage = page !== undefined ? page : currSearch.get('page');
  const newPerPage =
    perPage !== undefined ? perPage : currSearch.get('per_page');
  !isNaN(parseInt(newPage)) && searchParams.append('page', newPage);
  !isNaN(parseInt(newPerPage)) && searchParams.append('per_page', newPerPage);
};
