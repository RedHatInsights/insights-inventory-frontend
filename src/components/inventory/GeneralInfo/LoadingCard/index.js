export { default } from './LoadingCard';
export * from './LoadingCard';
