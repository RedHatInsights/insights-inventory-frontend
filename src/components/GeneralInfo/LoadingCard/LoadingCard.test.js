import '@testing-library/jest-dom';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { TestWrapper } from '../../../Utilities/TestingUtilities';
import LoadingCard, { Clickable } from './LoadingCard';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'localhost:3000/example/path',
  }),
  useParams: () => ({
    modalId: 'path',
  }),
}));

describe('LoadingCard', () => {
  [true, false].map((isLoading) => {
    it(`Loading card render - isLoading: ${isLoading}`, () => {
      const view = render(
        <TestWrapper>
          <LoadingCard
            isLoading={isLoading}
            title={`Card that is ${isLoading ? 'loading' : 'loaded'}`}
          />
        </TestWrapper>
      );

      expect(view.asFragment()).toMatchSnapshot();
    });
  });

  it('should render loading bars', () => {
    const view = render(
      <TestWrapper>
        <LoadingCard
          isLoading={true}
          title="Some title"
          items={[
            {
              onClick: jest.fn(),
              title: 'test-title',
              size: 'md',
              value: 'some value',
            },
            {
              title: 'just title',
            },
          ]}
        />
      </TestWrapper>
    );

    expect(view.asFragment()).toMatchSnapshot();
  });

  it(`Loading card render`, () => {
    const view = render(
      <TestWrapper>
        <LoadingCard
          isLoading={false}
          title="Some title"
          items={[
            {
              onClick: jest.fn(),
              title: 'test-title',
              size: 'md',
              value: 'some value',
            },
            {
              title: 'just title',
            },
          ]}
        />
      </TestWrapper>
    );

    expect(view.asFragment()).toMatchSnapshot();
  });

  it('Clickable should render - no data', () => {
    const view = render(
      <TestWrapper>
        <Clickable onClick={jest.fn()} />
      </TestWrapper>
    );
    expect(view.asFragment()).toMatchSnapshot();
  });

  describe('none/not available', () => {
    it(`should not be clickable when the value is 0`, () => {
      render(
        <TestWrapper>
          <LoadingCard
            isLoading={false}
            title="Some title"
            items={[
              {
                onClick: jest.fn(),
                title: 'test-title',
                value: 0,
              },
            ]}
          />
        </TestWrapper>
      );

      expect(screen.getByRole('definition')).toHaveTextContent(/^None$/);
      expect(screen.queryByRole('link')).not.toBeInTheDocument();
    });

    it(`should not be clickable when the value is 0 with plural`, () => {
      render(
        <TestWrapper>
          <LoadingCard
            isLoading={false}
            title="Some title"
            items={[
              {
                onClick: jest.fn(),
                title: 'test-title',
                value: 0,
                singular: 'system',
              },
            ]}
          />
        </TestWrapper>
      );

      expect(screen.getByRole('definition')).toHaveTextContent(/^0 systems$/);
      expect(screen.queryByRole('link')).not.toBeInTheDocument();
    });

    it(`should not be clickable when the value is undefined`, () => {
      render(
        <TestWrapper>
          <LoadingCard
            isLoading={false}
            title="Some title"
            items={[
              {
                onClick: jest.fn(),
                title: 'test-title',
                value: undefined,
              },
            ]}
          />
        </TestWrapper>
      );

      expect(screen.getByRole('definition')).toHaveTextContent(
        /^Not available$/
      );
      expect(screen.queryByRole('link')).not.toBeInTheDocument();
    });

    it(`should be none when value is 0`, () => {
      render(
        <TestWrapper>
          <LoadingCard
            isLoading={false}
            title="Some title"
            items={[
              {
                title: 'test-title',
                value: 0,
              },
            ]}
          />
        </TestWrapper>
      );

      expect(screen.getByRole('definition')).toHaveTextContent(/^None$/);
      expect(screen.queryByRole('link')).not.toBeInTheDocument();
    });

    it(`should be not available when value is undefined`, () => {
      render(
        <TestWrapper>
          <LoadingCard
            isLoading={false}
            title="Some title"
            items={[
              {
                title: 'test-title',
                value: undefined,
              },
            ]}
          />
        </TestWrapper>
      );

      expect(screen.getByRole('definition')).toHaveTextContent(
        /^Not available$/
      );
      expect(screen.queryByRole('link')).not.toBeInTheDocument();
    });

    it(`plurazied none`, () => {
      render(
        <TestWrapper>
          <LoadingCard
            isLoading={false}
            title="Some title"
            items={[
              {
                title: 'test-title',
                value: 0,
                singular: 'system',
              },
            ]}
          />
        </TestWrapper>
      );

      expect(screen.getByRole('definition')).toHaveTextContent(/^0 systems$/);
      expect(screen.queryByRole('link')).not.toBeInTheDocument();
    });

    it(`should be clickable with plural`, () => {
      render(
        <TestWrapper>
          <LoadingCard
            isLoading={false}
            title="Some title"
            items={[
              {
                onClick: jest.fn(),
                title: 'test-title',
                value: 23,
                singular: 'system',
              },
            ]}
          />
        </TestWrapper>
      );

      expect(screen.getByRole('definition')).toHaveTextContent(/^23 systems$/);
      expect(screen.getAllByRole('link')).toHaveLength(1);
    });

    it(`should be clickable with custom plural`, () => {
      render(
        <TestWrapper>
          <LoadingCard
            isLoading={false}
            title="Some title"
            items={[
              {
                onClick: jest.fn(),
                title: 'test-title',
                value: 23,
                singular: 'process',
                plural: 'processes',
              },
            ]}
          />
        </TestWrapper>
      );

      expect(screen.getByRole('definition')).toHaveTextContent(
        /^23 processes$/
      );
      expect(screen.getAllByRole('link')).toHaveLength(1);
    });
  });

  it('Clickable should render', () => {
    render(
      <TestWrapper>
        <Clickable value="15" target="some-target" />
      </TestWrapper>
    );

    expect(screen.getByRole('link', { name: /15/i })).toHaveAttribute(
      'href',
      '/some-target'
    );
  });

  it('clickable should click', async () => {
    const onClick = jest.fn();

    render(
      <TestWrapper>
        <Clickable onClick={onClick} value="15" target="path" />,
      </TestWrapper>
    );

    await userEvent.click(screen.getByRole('link', { name: /15/i }));
    await waitFor(() => {
      expect(onClick).toHaveBeenCalledTimes(1);
    });
  });

  it('Clickable should render - 0 value', async () => {
    const onClick = jest.fn();

    const view = render(
      <TestWrapper>
        <Clickable onClick={onClick} value={0} target="some-target" />
      </TestWrapper>
    );

    await waitFor(() => {
      expect(onClick).not.toHaveBeenCalled();
    });
    expect(view.asFragment()).toMatchSnapshot();
  });
});
