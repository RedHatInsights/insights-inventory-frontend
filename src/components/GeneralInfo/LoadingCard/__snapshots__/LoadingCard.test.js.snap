// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingCard Clickable should render - 0 value 1`] = `
<DocumentFragment>
  <a
    href="localhost:3000/example/path/some-target"
  >
    None
  </a>
</DocumentFragment>
`;

exports[`LoadingCard Clickable should render - no data 1`] = `
<DocumentFragment>
  <a
    href="localhost:3000/example/path/undefined"
  >
    Not available
  </a>
</DocumentFragment>
`;

exports[`LoadingCard Loading card render - isLoading: false 1`] = `
<DocumentFragment>
  <article
    class="pf-c-card"
    data-ouia-component-id="system-properties-card"
    data-ouia-component-type="PF4/Card"
    data-ouia-safe="true"
    id=""
  >
    <div
      class="pf-c-card__body"
    >
      <div
        class="pf-l-stack pf-m-gutter"
      >
        <div
          class="pf-l-stack__item"
        >
          <div
            class="pf-c-content"
          >
            <h1
              class=""
              data-ouia-component-id="SystemPropertiesCardTitle"
              data-ouia-component-type="PF4/Text"
              data-ouia-safe="true"
              data-pf-content="true"
            >
              Card that is loaded
            </h1>
          </div>
        </div>
        <div
          class="pf-l-stack__item pf-m-fill"
        />
      </div>
    </div>
  </article>
</DocumentFragment>
`;

exports[`LoadingCard Loading card render - isLoading: true 1`] = `
<DocumentFragment>
  <article
    class="pf-c-card"
    data-ouia-component-id="system-properties-card"
    data-ouia-component-type="PF4/Card"
    data-ouia-safe="true"
    id=""
  >
    <div
      class="pf-c-card__body"
    >
      <div
        class="pf-l-stack pf-m-gutter"
      >
        <div
          class="pf-l-stack__item"
        >
          <div
            class="pf-c-content"
          >
            <h1
              class=""
              data-ouia-component-id="SystemPropertiesCardTitle"
              data-ouia-component-type="PF4/Text"
              data-ouia-safe="true"
              data-pf-content="true"
            >
              Card that is loading
            </h1>
          </div>
        </div>
        <div
          class="pf-l-stack__item pf-m-fill"
        />
      </div>
    </div>
  </article>
</DocumentFragment>
`;

exports[`LoadingCard Loading card render 1`] = `
<DocumentFragment>
  <article
    class="pf-c-card"
    data-ouia-component-id="system-properties-card"
    data-ouia-component-type="PF4/Card"
    data-ouia-safe="true"
    id=""
  >
    <div
      class="pf-c-card__body"
    >
      <div
        class="pf-l-stack pf-m-gutter"
      >
        <div
          class="pf-l-stack__item"
        >
          <div
            class="pf-c-content"
          >
            <h1
              class=""
              data-ouia-component-id="SystemPropertiesCardTitle"
              data-ouia-component-type="PF4/Text"
              data-ouia-safe="true"
              data-pf-content="true"
            >
              Some title
            </h1>
          </div>
        </div>
        <div
          class="pf-l-stack__item pf-m-fill"
        >
          <div
            class="pf-c-content"
          >
            <dl
              class=""
            >
              <dt
                aria-label="test-title title"
                class=""
                data-ouia-component-id="test-title title"
              >
                test-title
              </dt>
              <dd
                aria-label="test-title value"
                class=""
                data-ouia-component-id="test-title value"
              >
                <div>
                  <a
                    href="localhost:3000/example/path/undefined"
                  >
                    some value
                  </a>
                </div>
              </dd>
              <dt
                aria-label="just title title"
                class=""
                data-ouia-component-id="just title title"
              >
                just title
              </dt>
              <dd
                aria-label="just title value"
                class=""
                data-ouia-component-id="just title value"
              >
                Not available
              </dd>
            </dl>
          </div>
        </div>
      </div>
    </div>
  </article>
</DocumentFragment>
`;

exports[`LoadingCard should render loading bars 1`] = `
<DocumentFragment>
  <article
    class="pf-c-card"
    data-ouia-component-id="system-properties-card"
    data-ouia-component-type="PF4/Card"
    data-ouia-safe="true"
    id=""
  >
    <div
      class="pf-c-card__body"
    >
      <div
        class="pf-l-stack pf-m-gutter"
      >
        <div
          class="pf-l-stack__item"
        >
          <div
            class="pf-c-content"
          >
            <h1
              class=""
              data-ouia-component-id="SystemPropertiesCardTitle"
              data-ouia-component-type="PF4/Text"
              data-ouia-safe="true"
              data-pf-content="true"
            >
              Some title
            </h1>
          </div>
        </div>
        <div
          class="pf-l-stack__item pf-m-fill"
        >
          <div
            class="pf-c-content"
          >
            <dl
              class=""
            >
              <dt
                aria-label="test-title title"
                class=""
                data-ouia-component-id="test-title title"
              >
                test-title
              </dt>
              <dd
                aria-label="test-title value"
                class=""
                data-ouia-component-id="test-title value"
              >
                <div
                  class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                >
                  <span
                    class="pf-u-screen-reader"
                  />
                </div>
              </dd>
              <dt
                aria-label="just title title"
                class=""
                data-ouia-component-id="just title title"
              >
                just title
              </dt>
              <dd
                aria-label="just title value"
                class=""
                data-ouia-component-id="just title value"
              >
                <div
                  class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
                >
                  <span
                    class="pf-u-screen-reader"
                  />
                </div>
              </dd>
            </dl>
          </div>
        </div>
      </div>
    </div>
  </article>
</DocumentFragment>
`;
